engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/BuiltinRootNode.java

```diff
@@ -33,66 +29,4 @@ protected BuiltinRootNode(EnsoLanguage language) {
    */
   @Override
   public abstract String getName();
-
-  /**
-   * Factory method creating a {@link DirectCallNode} to invoke this builtin.Defaults to standard
-   * {@link DirectCallNode#create(com.oracle.truffle.api.CallTarget)} implementation. Subclasses may
-   * override this with the help of {@link InlinedCallNode}.
-   *
-   * @return new node to use to call this builtin
-   */
-  public DirectCallNode createDirectCallNode() {
-    var callNode = DirectCallNode.create(cloneUninitialized().getCallTarget());
-    callNode.forceInlining();
-    return callNode;
-  }
-
-  /**
-   * Helper class allowing better implementation of {@link #createDirectCallNode}. Subclass, pass in
-   * {@code extra} and {@code body} and override {@code call} method to do what has to be done.
-   *
-   * @param <E> extra data to keep in the node
-   * @param <N> node to delegate to from {@link #call(java.lang.Object...)} method
-   */
-  protected abstract static class InlinedCallNode<E, N extends Node> extends DirectCallNode {
```

Comment:
We've been warned (in addition to the [javadoc](https://www.graalvm.org/truffle/javadoc/com/oracle/truffle/api/nodes/DirectCallNode.html) warning), that subclassing `DirectCallNode` creates unnecessary polymorphism when calling `DirectCallNode.call`. Such a polymorphism isn't a problem for _"PE mode"_  (the `Node` being invoke is always a compilation constant, so it is clear what `call` is going to be invoked), but before the JVM gets there, it slows things down (`Node` isn't compilation constant for the regular JVM).